metrics/config.go

```diff
@@ -147,15 +159,42 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
 // Record applies the `ros` Options to `ms` and then records the resulting
 // measurements in the metricsConfig's designated backend.
 func (mc *metricsConfig) Record(ctx context.Context, ms stats.Measurement, ros ...stats.Options) error {
+	var err error
 	if mc == nil || mc.recorder == nil {
+		ros, err = resourceToTags(ctx, ros...)
+		if err != nil {
+			return err
+		}
 		return stats.RecordWithOptions(ctx, append(ros, stats.WithMeasurements(ms))...)
 	}
 	return mc.recorder(ctx, ms, ros...)
 }
 
+// resourceToTags is a backwards-compatibility shim to forward all the labels
+// on the Resource context to tags (used for Prometheus and Stackdriver during
+// the transition to using Resources).
+func resourceToTags(ctx context.Context, ros ...stats.Options) ([]stats.Options, error) {
+	if r, ok := metricskey.FromContext(ctx); ok && r != nil {
+		tags := make([]tag.Mutator, 0, len(r.Labels))
+		for label, v := range r.Labels {
+			key, err := tag.NewKey(label)
+			if err != nil {
+				return nil, fmt.Errorf("bad resource label %q can't be converted to a tag: %+v", label, err)
```

Comment:
there's also `%w` change here :)
```suggestion
				return nil, fmt.Errorf("bad resource label %q can't be converted to a tag: %w", label, err)
```